File: BusinessLogic/Facturacion/Mapping/Detalle_Compra.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using APPCORE;
namespace DataBaseModel {
   

    public class Detalle_Compra : EntityClass
    {
        [PrimaryKey(Identity = true)]
        public int? Id_Detalle_Compra { get; set; }
        public int? Id_Compra { get; set; }
        public int? Id_Producto { get; set; }
        public double? Cantidad { get; set; }
        public double? Precio_Unitario { get; set; }
        public double? Precio_Venta { get; set; }
        public double? SubTotal { get; set; }
        public double? Iva { get; set; }
        public double? Total { get; set; }
        public string? Presentacion { get; set; }
        //Datos del producto que se agregara al lote
        public Transactional_Valoracion? Datos_Producto_Lote { get; set; }
        
        [ManyToOne(TableName = "Cat_Producto", KeyColumn = "Id_Producto", ForeignKeyColumn = "Id_Producto")]
        public Cat_Producto? Cat_Producto { get; set; }
        //[ManyToOne(TableName = "Tbl_Compra", KeyColumn = "Id_Compra", ForeignKeyColumn = "Id_Compra")]
        public Tbl_Compra? Tbl_Compra { get; set; }
        //[OneToMany(TableName = "Tbl_Lotes", KeyColumn = "Id_Detalle_Compra", ForeignKeyColumn = "Id_Detalle_Compra")]
        public List<Tbl_Lotes>? lotes { get; set; }
    }
}

File: UI/wwwroot/Facturacion/FrontModel/Detalle_Compra.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
import { Cat_Producto }  from './Cat_Producto.js'
import { Tbl_Compra }  from './Tbl_Compra.js'
import { Tbl_Lotes }  from './Tbl_Lotes.js'
class Detalle_Compra extends EntityClass {
   constructor(props) {
       super(props, 'EntityFacturacion');
       Object.assign(this, props);;
       this.Aplica_Iva = undefined;
       this.Precio_Venta = undefined;
   }
   /**@type {Number}*/ Id_Detalle_Compra;
   /**@type {Number}*/ Cantidad;
   /**@type {Number}*/ Precio_Unitario;
   /**@type {Number}*/ SubTotal;
   /**@type {Number}*/ Iva;
   /**@type {Number}*/ Total;
   /**@type {String}*/ Presentacion;   
   /**@type {Object}*/  Datos_Producto_Lote;
   /**@type {Cat_Producto} ManyToOne*/ Cat_Producto;
   /**@type {Tbl_Compra} ManyToOne*/ Tbl_Compra;
   /**@type {Array<Tbl_Lotes>} OneToMany*/ Tbl_Lotes;
}
export { Detalle_Compra }
